flink-runtime/src/test/java/org/apache/flink/runtime/jobmaster/slotpool/DeclarativeSlotPoolBridgeTest.java

```diff
@@ -55,13 +59,26 @@
 import static org.junit.Assert.fail;
 
 /** Tests for the {@link DeclarativeSlotPoolBridge}. */
+@RunWith(Parameterized.class)
 public class DeclarativeSlotPoolBridgeTest extends TestLogger {
 
     private static final Time rpcTimeout = Time.seconds(20);
     private static final JobID jobId = new JobID();
     private static final JobMasterId jobMasterId = JobMasterId.generate();
     private final ComponentMainThreadExecutor mainThreadExecutor =
             ComponentMainThreadExecutorServiceAdapter.forMainThread();
+    private final RequestSlotMatchingStrategy requestSlotMatchingStrategy;
+
+    @Parameterized.Parameters(name = "RequestSlotMatchingStrategy: {0}")
+    public static Collection<RequestSlotMatchingStrategy> data() throws IOException {
+        return Arrays.asList(
+                SimpleRequestSlotMatchingStrategy.INSTANCE,
+                PreferredAllocationRequestSlotMatchingStrategy.INSTANCE);
+    }
+
+    public DeclarativeSlotPoolBridgeTest(RequestSlotMatchingStrategy requestSlotMatchingStrategy) {
+        this.requestSlotMatchingStrategy = requestSlotMatchingStrategy;
```

Comment:
This tests that the different slot matching strategy doesn't break anything. Should we also test that it's really taken into account? (same with the `DeclarativeSlotPoolBridgeResourceDeclarationTest.java`)

Reply:
I am not sure whether it adds all that much additional benefit since we already have `PreferredAllocationRequestSlotMatchingStrategyTest` that tests the strategy. So what we are effectively testing is that the strategy is called. But maybe it is not too hard to add a test.

Reply:
You're right, I've missed that when this test couldn't pass without the `RequestSlotMatchingStrategy` being called. 👍